### PR TITLE
Add spaces_per_tab param to MD010

### LINKED ISSUE #384: Add ability to configure spaces to fix to in no-hard-tabs

### PR DESCRIPTION
**Pull request by Yash-Singh1**
Fixes #384

### CODE REVIEW

File: test/markdownlint-test-result-object.js [hunk 1]
```diff
@@ -346,7 +346,13 @@ test.cb("resultFormattingV3", (t) => {
         "# Heading   \n" +
         "\n" +
         "Text\ttext\t\ttext\n" +
-        "Text * emphasis * text"
+        "Text * emphasis * text\n" +
+        "\t\tInformation"
```
Comment (DavidAnson) [inline test/markdownlint-test-result-object.js:350]: Is there a reason to test it here? If not, it would be better (also easier) to add a new test file and use the "{MD010}" syntax. I'd want one for some number (4?) and another for 0. There should already be a set of files that verify the fix for this rule.

**Review by DavidAnson — CHANGES_REQUESTED** [review]
Thanks! I left a few comments and ideas for review. I like the idea.

File: test/markdownlint-test-result-object.js (hunk 1 above)
Comment (Yash-Singh1) [reply test/markdownlint-test-result-object.js:350]: > Is there a reason to test it here?

It was one of the tests on MD010, so I decided to use it.

> If not, it would be better (also easier) to add a new test file

Could you point to some examples? Even though I found multiple test files, I was not able understand where it was being tested.

> use the "{MD010}" syntax

What is this syntax? How would I tell the expected fixes using this syntax?

Comment (DavidAnson) [reply test/markdownlint-test-result-object.js:350]: Have a look at this set of files:
- https://github.com/DavidAnson/markdownlint/blob/main/test/detailed-results-html-tags.md
- https://github.com/DavidAnson/markdownlint/blob/main/test/detailed-results-html-tags.json
- https://github.com/DavidAnson/markdownlint/blob/main/test/detailed-results-html-tags.results.json
- https://github.com/DavidAnson/markdownlint/blob/main/test/detailed-results-html-tags.md.fixed

I prefer that style over changing code - the test you changed here is mainly about verifying the result object format - not rule behavior and fixes like the above format.

(My comment about "{MD010}" does not apply here, only to tests that do not verify fix output, sorry.)

Comment (DavidAnson) [reply test/markdownlint-test-result-object.js:350]: Also, let's test for the default (1), a typical value (4) and the edge case (0).

File: lib/md010.js [hunk 2]
```diff
@@ -14,6 +14,10 @@ module.exports = {
   "function": function MD010(params, onError) {
     const codeBlocks = params.config.code_blocks;
     const includeCodeBlocks = (codeBlocks === undefined) ? true : !!codeBlocks;
+    const spacesPerTab = params.config.spaces_per_tab;
+    const spaceMultiplier = (spacesPerTab === undefined) ?
+      1 :
+      Number(spacesPerTab);
```
Comment (DavidAnson) [inline lib/md010.js:20]: Math.max(0, Number(spacesPerTab))

**Review by DavidAnson — CHANGES_REQUESTED** [review]
Thanks for the updates - looking good! Answered your question and added one new comment.

File: test/markdownlint-test-result-object.js (hunk 1 above)
Comment (Yash-Singh1) [reply test/markdownlint-test-result-object.js:350]: 434edc9

File: test/markdownlint-test-result-object.js [hunk 3]
```diff
@@ -346,7 +346,13 @@ test.cb("resultFormattingV3", (t) => {
         "# Heading   \n" +
         "\n" +
         "Text\ttext\t\ttext\n" +
-        "Text * emphasis * text"
+        "Text * emphasis * text\n" +
```
Comment (DavidAnson) [inline test/markdownlint-test-result-object.js:349]: Could the changes to this file be reverted by adding a third set of test files that leave the default value of 1?

File: test/detailed-results-4-tabs-MD010.results.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+[
+  {
+    "errorContext": null,
+    "errorDetail": "Column: 1",
+    "errorRange": [1, 13],
+    "lineNumber": 1,
+    "ruleDescription": "Hard tabs",
+    "ruleInformation": "https://github.com/DavidAnson/markdownlint/blob/v0.23.1/doc/Rules.md#md010",
```
Comment (DavidAnson) [inline test/detailed-results-4-tabs-MD010.results.json:8]: Please update the version to "v0.0.0" in these places so they aren't outdated by version upgrades.

**Review by DavidAnson — CHANGES_REQUESTED** [review]
Nice! One test case change and one simple text replacement request remaining. Thank you!

**Review by DavidAnson — APPROVED** [review]
Great, thanks! I'll be rebasing this on to `next` branch in a moment.
